src/main/java/com/indico/Async.java

```diff
@@ -24,18 +44,31 @@
             }
         });
 
-        apolloCall.enqueue(new ApolloCall.Callback<T>() {
-            @Override
-            public void onResponse(@NotNull Response<T> response) {
-                completableFuture.complete(response);
-            }
+            try{
+            apolloCall.enqueue(new ApolloCall.Callback<T>() {
+                @Override
+                public void onResponse(@NotNull Response<T> response) {
+                    completableFuture.complete(response);
+                }
 
-            @Override
-            public void onFailure(@NotNull ApolloException e) {
-                completableFuture.completeExceptionally(e);
+                @Override
+                public void onFailure(@NotNull ApolloException e) {
+                    throw e;
+                }
+            });}
+            catch(ApolloNetworkException|ApolloHttpException e){
```

Comment:
We may want to make sure it's only for 504 and not all ApolloHttpExceptions. LIke regular 500s are probably not retryable.

Reply:
true, updated to be just for 504/503/502